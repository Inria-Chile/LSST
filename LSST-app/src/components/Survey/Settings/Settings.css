button.close {
    color: white;
}

.settings-button {
    position: absolute;
    top: 0;
    right: 0;
    background: var(--theme-active-slider);
    color: var(--theme-inactive-text);
    border: none;
    border-radius: 3px;
    margin: 0px;
    font-size: 25px;
    padding: 0px 10px;
    z-index: 3;
    outline: none !important;
    cursor: pointer;
}

.settings-wrapper {
    position: relative;
}

.sidebar {
    /* height: 60%; */
    display: none;
    width: 200px;
    position: absolute;
    padding: 15px;
    z-index: 1;
    top: 0px;
    right: 0;
    background: var(--theme-active-slider);
    border-radius: 10px;
    overflow-x: hidden;
    transition: 0.5s;
}

.sidebar label.selectable, .sidebar input {
    cursor: pointer;
}

.sidebar label, .sidebar input {
    -webkit-user-select: none; /* Safari */        
    -moz-user-select: none; /* Firefox */
    -ms-user-select: none; /* IE10+/Edge */
    user-select: none; /* Standard */
}

.sidebar-open {
    display: block;    
}
