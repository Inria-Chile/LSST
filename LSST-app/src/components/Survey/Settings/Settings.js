import React, { Component } from 'react';
// import ReactDOM from 'react-dom';
import './Settings.css';

class Settings extends Component {

    constructor(props) {
        super(props);
        /* establish the initial state of the input field */
        this.state = {
            showEcliptic: true,
            showGalactic: true,
            showMoon: true,
            showTelescopeRange: true,
            sidebarOpen: false,
            projection: "aitoff"
        }
        this.skymap = props.skymap;
    }

    toggleEcliptic = (e) => {
        console.log("toggleEcliptic", e);
        this.setState({showEcliptic: !this.state.showEcliptic});
        this.props.setEcliptic(!this.state.showEcliptic);
    }

    toggleGalactic = (e) => {
        console.log("toggleGalactic", e);
        this.setState({showGalactic: !this.state.showGalactic});
        this.props.setGalactic(!this.state.showGalactic);
    }

    toggleMoon = (e) => {
        console.log("toggleMoon", e);
        this.setState({showMoon: !this.state.showMoon});
        this.props.setMoon(!this.state.showMoon);
    }

    toggleTelescopeRange = (e) => {
        console.log("toggleTelescopeRange", e);
        this.setState({showTelescopeRange: !this.state.showTelescopeRange});
        this.props.setTelescopeRange(!this.state.showTelescopeRange);
    }

    toggleSettings = (e) => {
        this.props.setSettings(!this.state.sidebarOpen);
    }

    setProjection = (e) => {
        this.setState({projection: e.target.value});
        this.props.setProjection(e.target.value);
    }

    handleOutsideClick = (e) => {
        if (this.node.contains(e.target)) {
            return;
        }
        this.setState({
            sidebarOpen: false
        });
        document.removeEventListener('click', this.handleOutsideClick, false);
    }

    toggleSidebar = () => {
        if (!this.state.sidebarOpen) {
            document.addEventListener('click', this.handleOutsideClick, false);
        } else {
            document.removeEventListener('click', this.handleOutsideClick, false);
        }
        this.setState({
            sidebarOpen: !this.state.sidebarOpen
        })
    }

    

    render() {
        return (
            <div ref={node => { this.node = node; }}>
                <button className="settings-button" type="button" onClick={this.toggleSidebar} aria-label="Settings">
                    <i className="fa fa-cog" aria-hidden="true"></i>
                </button>
                <div className={"sidebar " + (this.state.sidebarOpen ? 'sidebar-open' : '')}>
                    <div className="checkbox">
                        <label className='selectable'>
                            <input type="checkbox" value="" onChange={this.toggleEcliptic} defaultChecked={this.state.showEcliptic}/> Ecliptic plane
                        </label>
                    </div>
                    <div className="checkbox">
                        <label className='selectable'>
                            <input type="checkbox" value="" onChange={this.toggleGalactic} defaultChecked={this.state.showGalactic}/> Galactic plane
                        </label>
                    </div>
                    <div className="checkbox">
                        <label className='selectable'>
                            <input type="checkbox" value="" onChange={this.toggleTelescopeRange} defaultChecked={this.state.showTelescopeRange}/> Elevation limit
                        </label>
                    </div>
                    <div className="checkbox">
                        <label className='selectable'>
                            <input type="checkbox" value="" onChange={this.toggleMoon} defaultChecked={this.state.showMoon}/> Moon
                        </label>
                    </div>
                    <label>
                        Projection: 
                        <select className="selectpicker" defaultValue= "aitoff" onChange={this.setProjection}>
                            <option>aitoff</option>
                            <option>armadillo</option>
                            <option>boggs</option>
                            <option>bromley</option>
                            <option>collignon</option>
                            <option>craster</option>
                            <option>cylindricalEqualArea</option>
                            <option>eckert1</option>
                            <option>eckert2</option>
                            <option>eckert3</option>
                            <option>eckert4</option>
                            <option>eckert5</option>
                            <option>eckert6</option>
                            <option>eisenlohr</option>
                            <option>equirectangular</option>
                            <option>foucaut</option>
                            <option>hammer</option>
                            <option>hatano</option>
                            <option>homolosine</option>
                            <option>mollweide</option>
                            <option>mtFlatPolarParabolic</option>
                            <option>nellHammer</option>
                            <option>robinson</option>
                            <option>sinusoidal</option>
                            <option>wagner4</option>
                            <option>wagner6</option>
                        </select>
                    </label>
                </div>
            </div>
        );
    }
}

export default Settings;
