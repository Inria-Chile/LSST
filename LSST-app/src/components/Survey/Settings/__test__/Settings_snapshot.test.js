import React from 'react';
import Settings from '../Settings';
import renderer from 'react-test-renderer';
import shallowRenderer from 'react-test-renderer/shallow';


describe('Settings snapshot test',function(){

    it('renders correctly with shallowRender',()=>{
        let setters = {
            setEcliptic: jest.fn(),
            setGalactic: jest.fn(),
            setMoon: jest.fn(),
            setTelescopeRange: jest.fn(),
            setProjection: jest.fn(),
        }
        const renderer = new shallowRenderer();
        const tree = renderer.render(
            <Settings {...setters} />
        );
        expect(tree).toMatchSnapshot();
    });
});
