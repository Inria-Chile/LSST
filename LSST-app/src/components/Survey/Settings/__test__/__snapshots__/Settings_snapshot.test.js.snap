// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings snapshot test renders correctly with shallowRender 1`] = `
<div>
  <button
    aria-label="Settings"
    className="settings-button"
    onClick={[Function]}
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-cog"
    />
  </button>
  <div
    className="sidebar "
  >
    skymap
    <div
      className="checkbox"
    >
      <label
        className="selectable"
      >
        <input
          defaultChecked={true}
          onChange={[Function]}
          type="checkbox"
          value=""
        />
         Ecliptic plane
      </label>
    </div>
    <div
      className="checkbox"
    >
      <label
        className="selectable"
      >
        <input
          defaultChecked={true}
          onChange={[Function]}
          type="checkbox"
          value=""
        />
         Galactic plane
      </label>
    </div>
    <div
      className="checkbox"
    >
      <label
        className="selectable"
      >
        <input
          defaultChecked={true}
          onChange={[Function]}
          type="checkbox"
          value=""
        />
         Elevation limit
      </label>
    </div>
    <div
      className="checkbox"
    >
      <label
        className="selectable"
      >
        <input
          defaultChecked={true}
          onChange={[Function]}
          type="checkbox"
          value=""
        />
         Moon
      </label>
    </div>
    <label>
      Projection:
      <select
        className="selectpicker"
        defaultValue="aitoff"
        onChange={[Function]}
      >
        <option>
          aitoff
        </option>
        <option>
          armadillo
        </option>
        <option>
          boggs
        </option>
        <option>
          bromley
        </option>
        <option>
          collignon
        </option>
        <option>
          craster
        </option>
        <option>
          cylindricalEqualArea
        </option>
        <option>
          eckert1
        </option>
        <option>
          eckert2
        </option>
        <option>
          eckert3
        </option>
        <option>
          eckert4
        </option>
        <option>
          eckert5
        </option>
        <option>
          eckert6
        </option>
        <option>
          eisenlohr
        </option>
        <option>
          equirectangular
        </option>
        <option>
          foucaut
        </option>
        <option>
          hammer
        </option>
        <option>
          hatano
        </option>
        <option>
          homolosine
        </option>
        <option>
          mollweide
        </option>
        <option>
          mtFlatPolarParabolic
        </option>
        <option>
          nellHammer
        </option>
        <option>
          robinson
        </option>
        <option>
          sinusoidal
        </option>
        <option>
          wagner4
        </option>
        <option>
          wagner6
        </option>
      </select>
    </label>
  </div>
</div>
`;
